src/gromacs/nblib/molecules.cpp

```diff
@@ -102,9 +102,20 @@ int Molecule::numAtomsInMolecule() const
     return atoms_.size();
 }
 
-void Molecule::addHarmonicBond(HarmonicType harmonicBond)
+int Molecule::numInteractionsInMolecule() const {
+    return interactionContainer_.interactions_.size();
+}
+
+size_t Molecule::getAtomIndex(AtomName atomName)
 {
-    harmonicInteractions_.push_back(harmonicBond);
+    for (size_t i = 0; i <= atoms_.size(); i++)
+    {
+        if (atoms_[i].atomName_ == atomName)
+        {
+            return i;
+        }
+    }
+    GMX_RELEASE_ASSERT(true, "Atom name not found. Please check again.");
```

Comment:
We can't have linear search because this leads to an N^2 algorithm.

Reply:
But it is linear in the number of types of bonds. In a single molecule/residue, that is a small handful.

Reply:
Yes, will work on a map based data structure to better represent this look-up scheme that isn't linear in price.